### DISCUSSION TITLE
How to config import path component from template Angular HTML

### DISCUSSION THREAD

**Discussion by namdh03** [community/community#137635] [Programming Help]
**How can I configure import paths in Angular HTML templates?**

I'm trying to automatically import using the path:

```typescript
import { ButtonComponent } from '~shared/components/button/button.component';
```

when I press `Ctrl + Space` in an HTML file. However, the result I get is:

```typescript
import { ButtonComponent } from '../../shared/components/button/button.component';
```
![Screenshot 2024-09-04 002852](https://github.com/user-attachments/assets/6f41618c-5401-4f87-9a36-4c739a615722)

Even though I have configured my `tsconfig.json` file like this:

```json
{
  "compileOnSave": false,
  "compilerOptions": {
    "outDir": "./dist/out-tsc",
    "strict": true,
    "noImplicitOverride": true,
    "noPropertyAccessFromIndexSignature": true,
    "noImplicitReturns": true,
    "noFallthroughCasesInSwitch": true,
    "skipLibCheck": true,
    "isolatedModules": true,
    "esModuleInterop": true,
    "sourceMap": true,
    "declaration": false,
    "experimentalDecorators": true,
    "moduleResolution": "bundler",
    "importHelpers": true,
    "target": "ES2022",
    "module": "ES2022",
    "lib": ["ES2022", "dom"],

    /* Paths */
    "baseUrl": "./src",
    "paths": {
      "~*": ["src/*"],
      "~core/*": ["app/core/*"],
      "~features/*": ["app/features/*"],
      "~shared/*": ["app/shared/*"]
    }
  },
  "angularCompilerOptions": {
    "enableI18nLegacyMessageIdFormat": false,
    "strictInjectionParameters": true,
    "strictInputAccessModifiers": true,
    "strictTemplates": true
  }
}
```

The extension I'm using is **Angular Language Service**.

![image](https://github.com/user-attachments/assets/647e19d5-27b8-44c7-abac-ece6f2172b4d)

But if I import the component manually like this:
![image](https://github.com/user-attachments/assets/b50bfa58-9c9f-4023-92ab-164ee05c6d9b)

Its working!

In short: I want to import path like that below when i use command ctrl + space in angular html file: 

```typescript
import { ButtonComponent } from '~shared/components/button/button.component';
```

**If anyone has encountered this problem before or knows how to fix it, your help would be invaluable.**

**Comment by iamdwn** [comment]
i have a same question
